ingest/Snakefile

```diff
@@ -101,45 +101,61 @@ rule clean:
 
 rule fetch_ncbi_dataset_package:
     output:
-        dataset_package="results/ncbi_dataset.zip",
+        dataset_package=directory("results/ncbi_dataset"),
     params:
         taxon_id=TAXON_ID,
         api_key=NCBI_API_KEY,
         gateway=f"--gateway-url {NCBI_GATEWAY_URL}" if NCBI_GATEWAY_URL else "",
         use_mirror=bool(MIRROR_BUCKET),
-        mirror_url=f"{MIRROR_BUCKET.strip('/')}/{TAXON_ID}.zip",
+        mirror_url=f"{MIRROR_BUCKET.strip('/')}/{TAXON_ID}.tar.zst",
+        unzip=unzip,
     shell:
         """
         if [[ "{params.use_mirror}" == "True" ]]; then
-            curl {params.mirror_url} -o {output.dataset_package}
+            curl {params.mirror_url} -o {output.dataset_package}.tar.zst
+            tar xvf {output.dataset_package}.tar.zst -C results
         else
             datasets download virus genome taxon {params.taxon_id} \
-            --no-progressbar \
-            --filename {output.dataset_package} \
-            --api-key {params.api_key} \
-            {params.gateway}
+                --no-progressbar \
+                --filename results/dataset.zip \
+                --api-key {params.api_key} \
+                {params.gateway} \
+            && {params.unzip} \
+                -o results/dataset.zip \
+                -d results
```

Comment:
Could you explain what this does differently?

Reply:
Is it that it does the extraction itself?

Reply:
Yeah because we now download a different format from the mirror we need different types of decompression depending on the type of data source used. Hence we do the extraction straight in the rule that does the download to keep things simple.